### PR TITLE
Terraform resource name refactor

### PR DESCRIPTION
**Pull request by aviraljain99**
I am not sure about all of these changes. But I tried to change resource names and attribute names to use underscores where possible.

### CODE REVIEW

**Review by harrykeightley — APPROVED** [review]
Looks good, approved with slight concern for the following https://cloud.google.com/compute/docs/naming-resources#resource-name-format  w.r.t.  resource names (might not apply if you got past the planning stage of terraform).

Note: as for wanting a linter for Terraform, I wasn't particularly happy with the VSCode extensions I tried, but you should be able to run terraform plan locally?

**Comment by aviraljain99** [comment]
> Looks good, approved with slight concern for the following https://cloud.google.com/compute/docs/naming-resources#resource-name-format w.r.t. resource names (might not apply if you got past the planning stage of terraform).
>
> Note: as for wanting a linter for Terraform, I wasn't particularly happy with the VSCode extensions I tried, but you should be able to run terraform plan locally?

So according to the link, GCP prefers using dashes rather than underscores?

**Comment by harrykeightley** [comment]
Only for resource naming I.e. the name field in the terraform resources

**Comment by aviraljain99** [comment]
@harrykeightley I see, I think I will stick to that convention then and change the names to use dashes. It is a bit unclear what needs to remain using dashes and what needs to remain using underscores. Because there are some attributes of resource blocks like the `display-name` attribute of the `google_api_gateway_api` block in the `api_gateway` module. 

What I will do is keep the attributes to use dashes rather than underscores and the actual resource block name to use underscores.

File: architecture/modules/functions/main.tf [hunk 1]
```diff
@@ -67,6 +67,9 @@ resource "google_cloudfunctions_function" "process_dataset" {
 }
 
 resource "google_cloudfunctions_function" "process_dataset_file" {
+  # ---- I am not sure if this name should be using dashes instead of underscores
+  # ---- According to the GCP terraform documentation, entry_point is the name of the
+  # ---- actual cloud function
   name        = "process_dataset_file"
```
Comment (nicklambourne) [inline architecture/modules/functions/main.tf:73]: I would leave towards dashes here, too, but not a huge problem.

**Review by nicklambourne — APPROVED** [review]
LGTM, you'll have to either delete and recreate a lot of this infra with the name changes (I suspect) or learn how to use `terraform state mv`. Let me know if you run into any dramas.

**Pull request merged**